nomad/structs/config/consul.go

```diff
@@ -260,6 +293,15 @@ func (c *ConsulConfig) Merge(b *ConsulConfig) *ConsulConfig {
 	if b.Namespace != "" {
 		result.Namespace = b.Namespace
 	}
+	if b.UseIdentity != nil {
+		result.UseIdentity = b.UseIdentity
+	}
+	if b.ServiceIdentity != nil {
+		result.ServiceIdentity = b.ServiceIdentity
+	}
+	if b.TemplateIdentity != nil {
+		result.TemplateIdentity = b.TemplateIdentity
+	}
```

Comment:
I don't immediately see other instances like this, but I think we would want to copy these values to prevent accidental changes?

Reply:
corrected in [7a0f84f](https://github.com/hashicorp/nomad/pull/18279/commits/7a0f84fabbbd2e6cb1a5efdb867d224fc6de1992)